doc/specs/#16599 - Quick Fix.md

```diff
@@ -0,0 +1,43 @@
+
+If and when the buffer resizes and reflows, the TerminalApp layer will be notified that the "contexts" are now invalidated. At that point, TerminalApp will call `GetQuickFixContext()` again to retrieve the updated information.
+
+The actual suggestions will be populated from various sources such as the following:
+- WinGet Command Not Found OSC Sequence:
+	- `OSC 9001; CmdNotFound; <missingCmd>` will be used by the attached CLI app to notify the terminal that a command `missingCmd` wasn't found. The TermControl layer will use WinGet's API to search for any packages that would remediate this error and construct a list of suggested packages to be installed.
+`OSC 9001` can be expanded on to add more suggestions via VT sequences from the underlying CLI app; for more on this, read the [Future considerations section](#future-considerations).
+## UI/UX Design
+A Quick Fix icon will be displayed immediately left of the prompt. If the icon fits in the left-gutter region, it will be drawn there. Otherwise, a cropped and transparent version will be displayed overlapping the text region; when the icon is hovered over, the icon will appear more clear by becoming more opaque and enlarging to become uncropped.
+
+When the icon (exposed as a button) is invoked, the Quick Fix menu (exposed as a flyout menu) will be displayed. This menu will include the following:
+- WinGet Command Not Found:
+	- If a command wasn't found in the immediately previous output (between prompts), this entry will be enabled. It includes a nested menu of all the WinGet suggestions. If no suggestions were found or the previous output didn't have a missing command, this entry will be disabled.
+- App Suggestions:
+	- The underlying CLI app could specifically recommend a command using a custom VT Sequence. For more information, read the [Future considerations section](#future-considerations).
+These suggestions will be displayed as "Run: \<command\>", which aligns with the design used in VS Code's Quick Fix menu.
+## Accessibility
+When a Quick Fix is available, an accessibility notification should be dispatched to alert the user that it is available. Aside from that, Quick Fix must be keyboard accessible via a default key binding. The UI will be a WinUI flyout menu, so it will be accessible by default once it is displayed and under focus.
+## Future considerations
+The Quick Fix menu can also be a home for future extensions to live. We can and should take inspiration from VS Code's implementation[^2][^3].
```

Comment:
You may also want to note that this paradigm could be used for more than just quick fixes, but also gutter marks for shell integration as well